lib/guest_installation_and_configuration_base.pm

```diff
@@ -0,0 +1,1565 @@
+#The subroutine can be called mainly in two different ways:
+#Firstly,config_guest_params can be called in another subroutine for example, create which takes a hash/dictionary as argument.
+#my %testhash = ('key1' => 'value1', 'key2' => 'value2', 'key3' => 'value3'),$self->create(%testhash) which calls $self->config_guest_params(@_).
+#Secondly,config_guest_params can also be called direcly, for example,$self->config_guest_params(%testhash).
+sub config_guest_params {
+    my $self          = shift;
+    my %_guest_params = @_;
+
+    $self->reveal_myself;
+    if ((scalar(@_) % 2 eq 0) and (scalar(@_) gt 0)) {
+        record_info("Configuring guest instance by using following parameters:", Dumper(\%_guest_params));
+        map { $self->{$_} = $_guest_params{$_} } keys %_guest_params;
+    }
+    else {
+        record_info("Can not configure guest instance with empty or odd number of arguments.Mark it as failed.", Dumper(\%_guest_params));
+        $self->record_guest_installation_result('FAILED') if ($self->{guest_installation_result} eq '');
+        return $self;
+    }
+
+    if ((defined $self->{guest_metadata}) and (grep { /name=/ } split(',', $self->{guest_metadata}))) {
+        my $_guest_name_in_metadata = grep { /name=(.*)/ } split(',', $self->{guest_metadata});
+        if (($self->{guest_name} ne $_guest_name_in_metadata) and ($self->{guest_installation_result} eq '')) {
+            record_info("The guest instance has a different name $_guest_name_in_metadata in metadata than $self->{guest_name}.", "Reset guest name to $_guest_name_in_metadata.");
+            $self->{guest_name} = $_guest_name_in_metadata;
+        }
+    }
+
+    if ($self->{guest_version} eq '') {
+        $self->{guest_version} = get_required_var('VERSION');
+        record_info("Guest $self->{guest_name} does not have guest_version set.Set it to test suite setting VERSION", "Please pay attention ! It is now $self->{guest_version}");
+    }
+
+    if (($self->{guest_version_major} eq '') or (!($self->{guest_version} =~ /^$self->{guest_version_major}.*(sp)?$/im))) {
+        ($self->{guest_version_major}) = $self->{guest_version} =~ /^(\d+)[-]?.*$/im;
+        record_info("Guest $self->{guest_name} does not have guest_version_major set or it does not match with guest_version.Set it from guest_version", "Please pay attention ! It is now $self->{guest_version_major}");
+    }
+
+    if (($self->{guest_version_minor} eq '') or (!($self->{guest_version} =~ /^(\d+)(-sp)?$self->{guest_version_minor}$/im))) {
+        $self->{guest_version} =~ /^.*(sp)(\d+)$/im;
+        $self->{guest_version_minor} = ($2 eq '' ? 0 : $2);
+        record_info("Guest $self->{guest_name} does not have guest_version_minor set or it does not match with guest_version.Set it from guest_version", "Please pay attention ! It is now $self->{guest_version_minor}");
+    }
+
+    if ($self->{guest_build} eq '') {
+        if ($self->{guest_version} eq lc get_required_var('VERSION')) {
```

Comment:
same with comment above. 'VERSION' here is host version.

Reply:
Repeat https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12363#discussion_r631491945. And please let me know if have better default value to set for guest_build if it is not given.

Reply:
I updated this part to including "TARGET_DEVELOPING_VERSION". So emtpy guest_version will be set to developing version anyway.